### PR TITLE
Fixed esp32 project link to http_parser

### PR DESCRIPTION
**Pull request by ericyeh1995**
Description
-----------

Fix http_parser link in esp32 project file. #1263 

https://amazon-freertos-ci.corp.amazon.com/job/master/job/custom/job/espressif/job/esp32_devkitc-ide/45/console

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/espressif/boards/esp32/aws_tests/application_code/espressif_code/amazon-freertos-tests/component.mk [hunk 1]
```diff
@@ -52,7 +52,7 @@ COMPONENT_SRCDIRS := $(AMAZON_FREERTOS_SDK_DIR)/standard/mqtt/src \
         $(AMAZON_FREERTOS_ABSTRACTIONS_DIR)/pkcs11/mbedtls \
         $(AMAZON_FREERTOS_ABSTRACTIONS_DIR)/platform/freertos \
         $(AMAZON_FREERTOS_ABSTRACTIONS_DIR)/secure_sockets/freertos_plus_tcp \
-        $(AMAZON_FREERTOS_3RD_PARTY_DIR)/http-parser \
+        $(AMAZON_FREERTOS_3RD_PARTY_DIR)/http_parser \
```
Comment (sarenameas) [inline vendors/espressif/boards/esp32/aws_tests/application_code/espressif_code/amazon-freertos-tests/component.mk]: We probably need to do this for the demo project also

Comment (ericyeh1995) [reply vendors/espressif/boards/esp32/aws_tests/application_code/espressif_code/amazon-freertos-tests/component.mk]: ok let me check

Comment (pavanmr94) [reply vendors/espressif/boards/esp32/aws_tests/application_code/espressif_code/amazon-freertos-tests/component.mk]: do a global search and replace in vendors directory

**Comment by ericyeh1995** [comment]
demo: https://amazon-freertos-ci.corp.amazon.com/job/master/job/custom/job/espressif/job/esp32_devkitc-ide/46/console

**Pull request merged**